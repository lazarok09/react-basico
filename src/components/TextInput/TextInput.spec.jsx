import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { TextInput } from "./index.jsx";

describe("<TextInput />", () => {
  it("should have a value on placeholder", () => {
    // cria função mock pro textInput
    const fn = jest.fn();

    render(<TextInput handleChange={fn} searchValue={"test"} />);

    // capitura o input pelo placeholder dele
    const input = screen.getByPlaceholderText(/type your search/i);
    // garante que o input está no documento
    expect(input).toBeInTheDocument();
    // confere o texto enviado via searchValue
    expect(input.value).toBe("test");
  });

  it("should call handleChange on each key is pressed", () => {
    // cria função mock pro textInput
    const fn = jest.fn();

    render(<TextInput handleChange={fn} />);
    // capitura o input
    const input = screen.getByPlaceholderText(/type your search/i);
    const typedValue = "the value to input";
    // digitar algo no input (valor acima)
    userEvent.type(input, typedValue);
    // verifica se o valor do input é o digitado acima via userEvent
    expect(input.value).toBe(typedValue);
    /*
    @ Vamos verificar agora se a função mock foi acionada a cada key pressionada
   */
    expect(fn).toHaveBeenCalledTimes(typedValue.length);
  });
  it("should match snapshot", () => {
    // cria função mock pro textInput
    const fn = jest.fn();
    // capitura o container pra fazer match snapshot
    const { container } = render(<TextInput handleChange={fn} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
