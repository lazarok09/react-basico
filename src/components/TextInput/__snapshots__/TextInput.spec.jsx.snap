// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match snapshot 1`] = `
<label
  for="search"
  id="label-search-container"
>
  <h1>
    Search
     : 
  </h1>
  <input
    id="search"
    placeholder="Type your search"
    type="search"
    value=""
  />
</label>
`;
